maven-api-impl/src/main/java/org/apache/maven/internal/impl/model/DefaultModelBuilder.java

```diff
@@ -595,6 +595,8 @@ Model transformFileToRaw(Model model) {
                     } else {
                         newDeps.add(dep);
                     }
+                } else {
+                    newDeps.add(dep);
```

Comment:
Because of this, the size of `newDeps` will always be equal to the number of objects in `model.getDependencies()`. We might as well pre-allocate the `ArrayList` with the correct size, so we avoid resizing `newDeps` while looping through `model.getDependencies()`.